### ISSUE TITLE
Custom Step No Longer Working After Switch to .json

### ISSUE THREAD

**Issue by LangLangBart** [continuedev/continue#632]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- Continue: 0.7.4 Pre-Release  2023-11-16T20:39:15.993Z
```

### Description

After updating to the latest version and transitioning to `.json`, my custom slash command (`/commit`) has stopped functioning. The error log indicates that `CommitMessageStep` is not recognized as a valid subclass of Step and it's not included in the list of permitted steps.

I also tested tit with the exact same code as outlined in the documentation [slash-commands.md#custom-slash-commands](https://github.com/continuedev/continue/blob/preview/docs/docs/customization/slash-commands.md#custom-slash-commands). This was functioning properly prior to the update.

I'm aware that the switch to `.json` is still a work in progress, and this issue might already be known internally. However, I wanted to properly report it to ensure it won't be forgotten.

### To reproduce

1. Write a custom step in your `~/.continue/config.py`, e.g. `CommitMessageStep`
```py
"""
Continue Code Configuration
"""

import subprocess
from continuedev.core.config import SlashCommand, ContinueConfig
from continuedev.core.main import ContinueCustomException, SetStep, Step
from continuedev.core.sdk import ContinueSDK

class CommitMessageStep(Step):
    async def run(self, sdk: ContinueSDK):
        # Get the root directory of the workspace
        dir = sdk.ide.workspace_directory

        # Run git diff in that directory
        diff = subprocess.check_output(
            ["git", "diff", "--staged"], cwd=dir).decode("utf-8")

        # Only run the command if diff is not empty
        if diff.strip():
            resp = await sdk.models.default.complete(
                f"{diff}\n\nPlease write a concise, specific commit message (less than 50 characters) about the changes above, following the Conventional Commits syntax in the format '<scope>: <message>'. Always include a scope. If the file changes consist only of markdown files, use 'docs' as the scope. If it's unclear, use 'chore' as the scope.")
            resp = resp.replace('\"', '').replace('\'', '').replace('`', '')
            await sdk.ide.runCommand(rf"git commit --message '{resp}'")
        else:
            raise ContinueCustomException(
                title="nothing to commit",
                message="If you want to commit, you need to add the changes to the staging area."
            )

def modify_config(config: ContinueConfig) -> ContinueConfig:
    config.slash_commands.append(
        SlashCommand(
            name="commit",
            description="Generate a commit message for the current changes",
            step=CommitMessageStep,
        )
    )
    return config
```

2. Open `VSCode` and `View Continue Server Logs`
3. Notice the error in the log messages

### Log output

```Shell
[2023-11-17 00:50:26] [WARNING] Failed to modify config: 2 validation errors for SlashCommand
step
  subclass of Step expected (type=type_error.subclass; expected_class=Step)
step
  unexpected value; permitted: 'AnswerQuestionChroma', 'GenerateShellCommandStep', 'EditHighlightedCodeStep', 'ShareSessionStep', 'CommentCodeStep', 'ClearHistoryStep', 'StackOverflowStep', 'OpenConfigStep' (type=value_error.const; given=CommitMessageStep; permitted=('AnswerQuestionChroma', 'GenerateShellCommandStep', 'EditHighlightedCodeStep', 'ShareSessionStep', 'CommentCodeStep', 'ClearHistoryStep', 'StackOverflowStep', 'OpenConfigStep'))
[2023-11-17 00:50:26] [DEBUG] ------ Begin Logs ------
[2023-11-17 00:50:27] [INFO] Meilisearch started at http://localhost:7700
[2023-11-17 00:50:28] [WARNING] Failed to modify config: 2 validation errors for SlashCommand
step
  subclass of Step expected (type=type_error.subclass; expected_class=Step)
step
  unexpected value; permitted: 'AnswerQuestionChroma', 'GenerateShellCommandStep', 'EditHighlightedCodeStep', 'ShareSessionStep', 'CommentCodeStep', 'ClearHistoryStep', 'StackOverflowStep', 'OpenConfigStep' (type=value_error.const; given=CommitMessageStep; permitted=('AnswerQuestionChroma', 'GenerateShellCommandStep', 'EditHighlightedCodeStep', 'ShareSessionStep', 'CommentCodeStep', 'ClearHistoryStep', 'StackOverflowStep', 'OpenConfigStep'))
[2023-11-17 00:50:29] [INFO] There are 1 sessions currently open
```

**Comment by sestinj** [comment]
Thank you for sharing this! Will be solving in the next pre-release. (via Union[StepName, str] instead of just StepName in pydantic model)

Aiming to fix all migration errors before moving .json changes to main, so keep the feedback coming if you find new things!

**Comment by sestinj** [comment]
@LangLangBart thanks for the report. Keep feedback coming! I'll have this solved in the next version, but want to be sure we get things right before making this migration available in the main version of the extension

**Comment by LangLangBart** [comment]
- The specific issue outlined here appears to have been resolved in `0.7.5 Pre-Release   2023-11-19T19:50:48.103Z ✹ 19/Nov/23`
	- The custom slash command now appears in the GUI dropdown.
- The issue was addressed with commit: c1c289fc.
```bash
git log --oneline --since=2023-11-16T20:39:15.993Z --until=2023-11-19T19:50:48.103Z -L :'class SlashCommand(BaseModel)':server/continuedev/core/config.py
```

```diff
--- a/server/continuedev/core/config.py
+++ b/server/continuedev/core/config.py
@@ -41,23 +41,23 @@
 class SlashCommand(BaseModel):
     name: str
     description: str
-    step: Union[Type[Step], StepName]
+    step: Union[Type[Step], StepName, str]
     params: Optional[Dict] = {}

     # Allow step class for the migration
     @validator("step", pre=True, always=True)
     def step_is_string(cls, v):
         if isinstance(v, str):
             return v
         elif isinstance(v, object) and v.__class__.__name__ == "ModelMetaclass":
             return str(v).split(".")[-1].split("'")[0]
         else:
             return v

     def slash_command_description(self) -> SlashCommandDescription:
         return SlashCommandDescription(
             name=self.name,
             description=self.description,
         )
```

---

> [!WARNING]
> However, when the custom slash command is executed, it does not perform the
> programmed steps defined in my `~/.continue/config.py`. Instead, it simply takes the entire slash
> command literally and does not resolve it to its defined function. Would you like me to start a
> new bug report, or should we keep this one open, @sestinj?

**Comment by sestinj** [comment]
@LangLangBart I found the issue and tested with your exact config.py to make sure it works. Already published in v0.7.22 and on the way for v0.6.8

**Comment by LangLangBart** [comment]
Everything is now working correctly. Commit f1acf952 must have contained the necessary fixes.
